WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/InPersonPaymentsMenuViewController.swift

```diff
@@ -418,6 +421,14 @@ private extension InPersonPaymentsMenuViewController {
             self?.configureSections(shouldShowTapToPayOnIPhoneFeedback: shouldShowFeedbackRow)
             self?.tableView.reloadData()
         }.store(in: &cancellables)
+
+        viewModel.$shouldBadgeTapToPayOnIPhone.sink { [weak self] _ in
+            self?.configureSections()
+            // ensures that the cell will be configured with the correct value for the badge
+            DispatchQueue.main.async {
+                self?.tableView.reloadData()
+            }
+        }.store(in: &cancellables)
```

Comment:
I don't think is necessary as we're already weakifying self, but should we add a guard just to remove optionality?
```suggestion
viewModel.$shouldBadgeTapToPayOnIPhone.sink { [weak self] _ in
    guard let self = self else { return }
    self.configureSections()
    DispatchQueue.main.async {
        self.tableView.reloadData()
    }
}.store(in: &cancellables)

```

Reply:
I don't think there's a benefit to removing the optionality here, but thanks for the suggestion 😊